### PR TITLE
examples: Add dynamic configuration (filesystem) sandbox

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: examples: Add dynamic configuration (filesystem) sandbox
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: examples/dynamic-config-fs/verify.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+responds_with \
+    "Request served by service2" \
+    http://localhost:10000
+
+run_log "Check config for active clusters pointing to service2"
+curl -s http://localhost:19000/config_dump \
+    | jq -r '.configs[1].dynamic_active_clusters' \
+    | grep '"address": "service2"'
+
+# curl -s http://localhost:19000/config_dump  \
```
Comment (phlax) [inline examples/dynamic-config-fs/verify.sh:38]: not sure why this doesnt work.

seems like the cluster gets updated, but the version doesnt

**Comment by phlax** [comment]
docs are here - http://storage.googleapis.com/envoy-pr/13783/docs/start/sandboxes/dynamic-configuration-filesystem.html

its still pretty wip

File: docs/root/start/sandboxes/dynamic-configuration-filesystem.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+.. include:: _include/docker-env-setup.rst
+
+Change directory to ``examples/dynamic-config-fs`` in the Envoy repository.
+
+Step 3: Start the proxy container
+*********************************
+
+.. note::
+
+   If you are running on a system with strict ``umask`` you will need to ``chmod`` the dynamic config
```
Comment (phlax) [inline docs/root/start/sandboxes/dynamic-configuration-filesystem.rst:20]: im not over-happy with this, but after thinking about it quite a bit, it seems like the least bad option to handle the `umask` issue

File: examples/dynamic-config-fs/verify.sh (hunk 1 above)
Comment (phlax) [reply examples/dynamic-config-fs/verify.sh:37]: seems like versions are not important here - at least for this sandbox

File: docs/root/start/sandboxes/dynamic-configuration-filesystem.rst [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+**************************
+
+You should be able to make a request to port ``10000``, which will be served by ``service1``.
+
+.. code-block:: console
+
+   $ curl -s http://localhost:10000
+   Request served by service1
+
+   HTTP/2.0 GET /
```
Comment (phlax) [inline docs/root/start/sandboxes/dynamic-configuration-filesystem.rst:60]: im confused as to why this is getting `HTTP/2.0` but the control plane sandbox, didnt

File: examples/BUILD [hunk 4]
```diff
@@ -18,6 +18,7 @@ filegroup(
         ],
         exclude = [
             "cache/responses.yaml",
+            "dynamic-config-fs/**/*",
```
Comment (phlax) [inline examples/BUILD:21]: this isnt ideal, but theres not an obvious way to include the dynamic `yaml` without it being treated as a standalone envoy config

File: .azure-pipelines/pipelines.yml [hunk 5]
```diff
@@ -167,7 +167,7 @@ stages:
               AZP_BRANCH: $(Build.SourceBranch)
 
   - stage: linux_x64
-    dependsOn: ["precheck"]
+    dependsOn: []
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:170]: revert

File: examples/dynamic-config-fs/README.md [hunk 6]
```diff
@@ -0,0 +1,2 @@
+To learn about this sandbox and for instructions on how to run it please head over
+to the [Envoy docs](https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/postgres.html).
```
Comment (phlax) [inline examples/dynamic-config-fs/README.md:2]: fix this

File: examples/dynamic-config-fs/configs/cds.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+version_info: "1"
```
Comment (phlax) [inline examples/dynamic-config-fs/configs/cds.yaml:1]: is this necessary?

Comment (phlax) [reply examples/dynamic-config-fs/configs/cds.yaml:1]: seems not removed

File: examples/dynamic-config-fs/README.md (hunk 6 above)
Comment (phlax) [reply examples/dynamic-config-fs/README.md:2]: fixed

**Comment by phlax** [comment]
this one should be ready for final review

File: docs/root/start/sandboxes/dynamic-configuration-filesystem.rst (hunk 3 above)
Comment (mattklein123) [reply docs/root/start/sandboxes/dynamic-configuration-filesystem.rst:60]: Hmm no clue.

File: examples/BUILD (hunk 4 above)
Comment (mattklein123) [reply examples/BUILD:21]: Hmm yeah. I think we could probably config check these separately somehow but not a huge deal for now.

**Review by mattklein123 — APPROVED** [review]
Awesome!

**Pull request merged**